File: 1768. Merge Strings Alternately.java

class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb = new StringBuilder();
        int min = Math.min(word1.length(), word2.length());
        for(int i = 0; i<=min; i++)
        {
            if(i == min)
            {
                if(word1.length() == word2.length()) break;

                if(word1.length() == min)
                    sb.append(word2.substring(min, word2.length()));
                else sb.append(word1.substring(min, word1.length()));
                
                break;
            }
            sb.append(word1.charAt(i));
            sb.append(word2.charAt(i));
        }
        return sb.toString();
    }
}

File: 1768. Merge Strings Alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        min_ = min(len(word1), len(word2))
        ans = ""
        for i in range(min_):
            ans += word1[i]
            ans += word2[i]

        if len(word1) > len(word2):
            ans += word1[min_:len(word1)]
        elif len(word2) > len(word1):
                ans += word2[min_:len(word2)]

        return ans

        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
